### PR TITLE
[base] moving to buster slim

### PR DESCRIPTION
**Pull request by truthbk**
*Note: Please remember to review the Datadog [Contribution Guidelines](https://github.com/DataDog/dd-agent/blob/master/CONTRIBUTING.md)
if you have not yet done so.*

### What does this PR do?

Updates the docker agent base image to use the more current buster slim

### Motivation

More current, secure base image.

### Testing Guidelines

An overview on [testing](https://github.com/DataDog/dd-agent/blob/master/tests/README.md)
is available in our contribution guidelines.

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -18,6 +18,9 @@ ENV DOCKER_DD_AGENT=yes \
     DD_SUPERVISOR_DELETE_USER=yes \
     DD_CONF_PROCFS_PATH="/host/proc"
 
+# workaround for stretch-slim missing man dirs
+RUN seq 1 8 | xargs -I{} mkdir -p /usr/share/man/man{}
```
Comment (olivielpeau) [inline Dockerfile:22]: I'm curious, why is this necessary?

Comment (truthbk) [reply Dockerfile:22]: I actually have to check if it's still necessary with `buster` - on slim those directories are missing, I left it in there because it's a NOP if the directory structure is in place. Not having it in place was breaking the build with `stretch-slim`, it was a well documented issue.

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM debian:stretch
+FROM debian:buster-20200607-slim
```
Comment (vboulineau) [inline Dockerfile:1]: Why fixed date base? Using `debian:buster-slim` would be more aligned with what we do for Agent7

Comment (truthbk) [reply Dockerfile:1]: Reproducible builds. I personally think this is better. But if you have a strong opinion let me know, we can fix in a follow-up PR.
